alibi_detect/utils/pytorch/kernels.py

```diff
@@ -1,8 +1,44 @@
+from abc import abstractmethod
 import numpy as np
 import torch
 from torch import nn
 from . import distance
-from typing import Optional, Union, Callable
+from typing import Optional, Union, Callable, List
+from copy import deepcopy
+
+
+def infer_kernel_parameter(
+    kernel: 'BaseKernel',
```

Comment:
Type can just be changed to `BaseKernel`? @jklaise can you confirm?

Reply:
Yea I can't see why not (perhaps an ordering issue as `BaseKernel` is defined further down?).

Reply:
Yes indeed, as it is defined below, mypy would suggest an error without the quote.